src/ImageSharp/Common/ParallelUtils/ParallelHelper.cs

```diff
@@ -142,5 +144,18 @@ public static void IterateRows(Rectangle rectangle, Configuration configuration,
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static int DivideCeil(int dividend, int divisor) => 1 + ((dividend - 1) / divisor);
+
+        private static void ValidateRectangle(Rectangle rectangle)
+        {
+            Guard.IsTrue(
+                rectangle.Width > 0,
+                $"{nameof(rectangle)}.{nameof(rectangle.Width)}",
+                "Can't iterate rows on a with a rectangle that has a width less than or equal to 0");
```

Comment:
We have `Guard.MustBeGreaterThan` which will throw the correct `ArgumentOutOfRangeException`. I don't think we need a custom message then.

Reply:
Ok I'll revert this to a guard